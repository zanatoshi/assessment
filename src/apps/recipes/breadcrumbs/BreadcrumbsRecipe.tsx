import {Breadcrumbs, createStyles, makeStyles, Theme} from '@material-ui/core';
import NavigateNextIcon from '@material-ui/icons/NavigateNext';
import React from 'react';
import {LinkRouter} from '../../../components/link-router/LinkRouter';

interface BreadcrumbsRecipeProps {
  meal?: string;
  course?: string;
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    breadcrumbs: {
      textTransform: 'uppercase',
      fontWeight: theme.typography.fontWeightMedium,
      color: theme.palette.common.black,
      marginBottom: theme.spacing(2),
    },
    separator: {
      color: theme.palette.secondary.dark,
    },
  })
);

export const BreadcrumbsRecipe = ({meal, course}: BreadcrumbsRecipeProps) => {
  const classes = useStyles();
  const mealParam = meal ? `?meal=${meal}` : '';
  return (
    <Breadcrumbs
      className={classes.breadcrumbs}
      separator={<NavigateNextIcon fontSize="small" className={classes.separator} />}
      aria-label="breadcrumb"
    >
      <LinkRouter color="inherit" href="/recipes">
        RECIPES
      </LinkRouter>
      {meal && (
        <LinkRouter color="inherit" href={`/recipes/meals/${meal}`}>
          {meal.replace('_', ' ').toUpperCase()}
        </LinkRouter>
      )}
      {course && (
        <LinkRouter color="inherit" href={`/recipes/courses/${course}${mealParam}`}>
          {course.replace('_', ' ').toUpperCase()}
        </LinkRouter>
      )}
    </Breadcrumbs>
  );
};
