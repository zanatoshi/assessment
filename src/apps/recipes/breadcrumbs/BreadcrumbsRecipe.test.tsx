import {cleanup, screen} from '@testing-library/react';
import React from 'react';
import {render} from '../../../test/test-utils';
import {BreadcrumbsRecipe} from './BreadcrumbsRecipe';

afterEach(() => {
  cleanup();
});

describe('Breadcrumbs recipe test', () => {
  const meal = 'test-meal';
  const course = 'test-course';
  it('Show the breadcrumbs without meal and course', () => {
    render(<BreadcrumbsRecipe />);
    const recipesLink = screen.getByText('RECIPES');
    expect(recipesLink).toBeTruthy();
    expect(screen.queryByText(meal.toUpperCase())).toBeNull();
    expect(screen.queryByText(course.toUpperCase())).toBeNull();
  });
  it('Show the breadcrumbs with meal and without course', () => {
    render(<BreadcrumbsRecipe meal={meal} />);
    expect(screen.getByText('RECIPES')).toBeTruthy();
    expect(screen.getByText(meal.toUpperCase())).toBeTruthy();
    expect(screen.queryByText(course.toUpperCase())).toBeNull();
  });
  it('Show the breadcrumbs with meal and course', () => {
    render(<BreadcrumbsRecipe meal={meal} course={course} />);
    expect(screen.getByText('RECIPES')).toBeTruthy();
    expect(screen.getByText(meal.toUpperCase())).toBeTruthy();
    expect(screen.getByText(course.toUpperCase())).toBeTruthy();
  });
  it('matches snapshot', () => {
    const {asFragment} = render(<BreadcrumbsRecipe meal={meal} course={course} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
