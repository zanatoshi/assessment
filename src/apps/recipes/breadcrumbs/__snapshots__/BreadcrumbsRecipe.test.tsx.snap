// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs recipe test matches snapshot 1`] = `
<DocumentFragment>
  <nav
    aria-label="breadcrumb"
    class="MuiTypography-root MuiBreadcrumbs-root makeStyles-breadcrumbs MuiTypography-body1 MuiTypography-colorTextSecondary"
  >
    <ol
      class="MuiBreadcrumbs-ol"
    >
      <li
        class="MuiBreadcrumbs-li"
      >
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
          href="/recipes"
        >
          RECIPES
        </a>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-separator MuiSvgIcon-fontSizeSmall"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
        </svg>
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
          href="/recipes/meals/test-meal"
        >
          TEST-MEAL
        </a>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-separator MuiSvgIcon-fontSizeSmall"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
        </svg>
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
          href="/recipes/courses/test-course?meal=test-meal"
        >
          TEST-COURSE
        </a>
      </li>
    </ol>
  </nav>
</DocumentFragment>
`;
